### PR TITLE
feat: mvtx gauge metrics support

### PR DESCRIPTION
**Pull request by veds-g**
image

### CODE REVIEW

**Review by vigith — APPROVED** [review]

File: config/base/numaflow-server/numaflow-server-metrics-proxy-config.yaml [hunk 1]
```diff
@@ -9,6 +9,33 @@ data:
     # example for local prometheus service
     # url: http://prometheus-operated.monitoring.svc.cluster.local:9090
     patterns:
+    - name: mono_vertex_gauge
+      object: mono-vertex
+      title: Pending Messages Lag
+      description: This query is the total number of pending messages for the mono vertex
+      expr: |
+        $metric_name{$filters}
```
Comment (veds-g) [inline config/base/numaflow-server/numaflow-server-metrics-proxy-config.yaml:17]: will be adding support for operation like max, min or avg in query expression.

**Review by whynowy — APPROVED** [review]

**Review by yhl25 — COMMENTED** [review]
codegen is failing, also please add unit tests to cover gauge metrics.

File: ui/src/components/pages/Pipeline/partials/Graph/partials/NodeInfo/partials/Pods/partials/PodDetails/partials/Metrics/partials/common/FiltersDropdown/index.tsx [hunk 2]
```diff
@@ -29,6 +29,13 @@ export interface FiltersDropdownProps {
   setFilters: any;
 }
 
+const periodData = [
```
Comment (adarsh0728) [inline ui/src/components/pages/Pipeline/partials/Graph/partials/NodeInfo/partials/Pods/partials/PodDetails/partials/Metrics/partials/common/FiltersDropdown/index.tsx:32]: periods of "1m", "5m" etc are defined at the backend?

**Review by adarsh0728 — COMMENTED** [review]
you can use **make manifests** so that changes are reflected in other configs. That may resolve codegen issue

File: ui/src/components/pages/Pipeline/partials/Graph/partials/NodeInfo/partials/Pods/partials/PodDetails/partials/Metrics/partials/common/FiltersDropdown/index.tsx (hunk 2 above)
Comment (adarsh0728) [reply ui/src/components/pages/Pipeline/partials/Graph/partials/NodeInfo/partials/Pods/partials/PodDetails/partials/Metrics/partials/common/FiltersDropdown/index.tsx:32]: found it https://github.com/numaproj/numaflow/blob/11c0d2b40c314d81517fe8afeea30edd50f3cda4/rust/numaflow-core/src/metrics.rs#L841

**Review by adarsh0728 — APPROVED** [review]

**Pull request merged**